test/unit/Locking/Redis/RedisDistributedAppLockTest.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Threading.Tasks;
+using FakeItEasy;
+using RapidCore.Locking;
+using RapidCore.Locking.Redis;
+using StackExchange.Redis;
+using Xunit;
+
+namespace RapidCore.UnitTests.Locking.Redis
+{
+    public class RedisDistributedAppLockTest
+    {
+        [Fact]
+        public void Does_acquire_lock_using_redisclient()
+        {
+            var lockName = "the-lock";
+            var client = A.Fake<IDatabase>(o => o.Strict());
+            var manager = A.Fake<IConnectionMultiplexer>(o => o.Strict());
+            A.CallTo(() => manager.GetDatabase(A<int>.Ignored, A<object>.Ignored)).Returns(client);
+            A.CallTo(() => client.LockTake(
+                A<RedisKey>.That.Matches(str => str == lockName),
+                A<RedisValue>.Ignored,
+                A<TimeSpan>.Ignored,
+                A<CommandFlags>.Ignored)).Returns(true);
+
+            var handle = new RedisDistributedAppLock(manager);
+
+            handle.AcquireLockAsync(lockName);
+
+            A.CallTo(() => client.LockTake(
+                    A<RedisKey>.That.Matches(str => str == lockName),
+                    A<RedisValue>.Ignored,
+                    A<TimeSpan>.Ignored,
+                    A<CommandFlags>.Ignored))
+                .MustHaveHappened();
+
+            Assert.Equal(lockName, handle.Name);
+        }
+
+        [Fact]
+        public void Does_acquire_lock_using_redisclient_w_timeout()
+        {
+            var lockName = "the-lock";
+
+            var client = A.Fake<IDatabase>();
+            var manager = A.Fake<IConnectionMultiplexer>();
+            A.CallTo(() => manager.GetDatabase(A<int>.Ignored, A<object>.Ignored)).Returns(client);
+            A.CallTo(() => client.LockTake(
+                A<RedisKey>.That.Matches(str => str == lockName),
+                A<RedisValue>.Ignored,
+                A<TimeSpan>.Ignored,
+                A<CommandFlags>.Ignored)).Returns(true);
+
+            var handle = new RedisDistributedAppLock(manager);
+            handle.AcquireLockAsync(lockName, TimeSpan.FromSeconds(2));
+
+            A.CallTo(() => client.LockTake(
+                    A<RedisKey>.That.Matches(str => str == lockName),
+                    A<RedisValue>.Ignored,
+                    A<TimeSpan>.Ignored,
```

Comment:
That timespan is the lock auto expiry - test is invalid after moving to StackExchange.Redis